GANDLF/cli/generate_metrics.py

```diff
@@ -41,27 +74,83 @@ def generate_metrics_dict(
         input_csv (str): The input CSV.
         config (str): The input yaml config.
         outputfile (str, optional): The output file to save the metrics. Defaults to None.
+        missing_prediction (int, optional): The value to use for missing predictions as penalty. Default is -1.
 
     Returns:
         dict: The metrics dictionary.
     """
-    input_df = pd.read_csv(input_csv)
+    # the case where the input is a comma-separated 2 files with targets and predictions
+    if "," in input_csv:
+        target_csv, prediction_csv = input_csv.split(",")
+        target_df = pd.read_csv(target_csv)
+        prediction_df = pd.read_csv(prediction_csv)
+        ## start sanity checks
+        # if missing predictions are not to be penalized, check if the number of rows in the target and prediction files are the same
+        if missing_prediction == -1:
+            assert (
+                target_df.shape[0] == prediction_df.shape[0]
+            ), "The number of rows in the target and prediction files should be the same"
+
+        # check if the number of columns in the target and prediction files are the same
+        assert (
+            target_df.shape[1] == prediction_df.shape[1]
+        ), "The number of columns in the target and prediction files should be the same"
+        assert (
+            target_df.shape[1] == 2
+        ), "The target and prediction files should have *exactly* 2 columns"
+
+        # find the correct header for the subjectID column
+        target_df = __update_header_location_case_insensitive(target_df, "SubjectID")
+        prediction_df = __update_header_location_case_insensitive(
+            prediction_df, "SubjectID"
+        )
+        # check if prediction_df has extra subjectIDs
+        assert (
+            prediction_df["SubjectID"].isin(target_df["SubjectID"]).all()
+        ), "The `SubjectID` column in the prediction file should be a subset of the `SubjectID` column in the target file"
+
+        # individual checks for target and prediction dataframes
+        for df in [target_df, prediction_df]:
+            # check if the "subjectID" column has duplicates
+            assert (
+                df["SubjectID"].duplicated().sum() == 0
+            ), "The `SubjectID` column should not have duplicates"
+
+            # check if SubjectID is the first column
+            assert (
+                df.columns[0] == "SubjectID"
+            ), "The `SubjectID` column should be the first column in the target and prediction files"
+
+        # change the column name after subjectID to target and prediction
+        target_df = target_df.rename(columns={target_df.columns[1]: "Target"})
+        prediction_df = prediction_df.rename(
+            columns={prediction_df.columns[1]: "Prediction"}
+        )
+
+        # combine the two dataframes
+        input_df = target_df.merge(prediction_df, how="left", on="SubjectID").fillna(
+            missing_prediction
+        )
```

Comment:
Just FYI - when a column has `NaN` values, the column dtype converts to a float, e.g.

```python
>>> target_df = pd.DataFrame({
...     "SubjectID": ["a", "b", "c", "d"],
...     "Target": np.random.randint(2, size=4)
... })
>>> target_df
  SubjectID  Target
0         a       0
1         b       1
2         c       1
3         d       0
>>> 
>>> prediction_df = pd.DataFrame({
...     "SubjectID": ["a", "b"],
...     "Prediction": np.random.randint(2, size=2)
... })
>>> prediction_df
  SubjectID  Prediction
0         a           1
1         b           0
>>> 
>>> missing_prediction = 2
>>> target_df.merge(prediction_df, how="left", on="SubjectID").fillna(missing_prediction)
  SubjectID  Target  Prediction
0         a       0         1.0
1         b       1         0.0
2         c       1         2.0
3         d       0         2.0
```

Not sure if this matters downstream, but just thought I'd note it, since I saw in the [docs](https://docs.mlcommons.org/GaNDLF/usage/#generate-metrics) that integers are used for classification and floats for regression.

Reply:
I did a quick check and it seemed to produce numbers as expected.

Reply:
Just tested it on my end as well and I also verify that the results are as expected 👍🏼

Reply:
Brilliant, resolving.